### ISSUE TITLE
Inefficient use of heap causes frequent gc

### ISSUE THREAD

**Issue by Nik-Gleb** [koral--/android-gif-drawable#367]
I like this library, thanks to the authors for their labor ;)
And soory for my bad english.

In my project i have grid of gif's (about 4x5, may be less) and today I was faced with frequent "gc-calls", and ever-growing using-heap. I discovered an issue in draw-method of gif-drawable.
`mRenderTaskSchedule = mExecutor.schedule(mRenderTask, renderDelay, TimeUnit.MILLISECONDS);`
Unfortunaly, java's native ScheduledThreadPoolExexcutor.schedule creates new objects, wrappers,...etc ever call

I think it makes sense to make the mRenderTaskSchedule reusable, or create any EfficientScheduledExecutor implementation, via reflections, may be jni reflections or other workarounds

Thanks.

**Comment by koral--** [comment]
Indeed, `ScheduledThreadPoolExecutor#schedule()` causes allocations.
Some classes from this library eg. `GifTexImage2D` use native (pthread) threads to change GIF frames.
It should be possible to use that mechanism also in `GifDrawable`. I'll check this.

**Comment by Nik-Gleb** [comment]
Ok. Perhaps it will be more cleanest solution. Thanks.

**Comment by koral--** [comment]
Assuming that each `GifDrawable` will spawn its own thread. It seems that it should be additional option, not replacement of current behavior.
Adding 2.x milestone for this issue.

**Comment by Nik-Gleb** [comment]
Hi!

You can also use the Handler-based approach, to schedule decode operations.
Android already knows how to do it correctly and there is a message pool for more efficient memory using.

I think, the drawable can be free from liability by threads-managing.
The user can specify his own looper for decoding operations, else MainLopper will be used by default.
[mHandler = new Handler(looper);](https://developer.android.com/reference/android/os/Handler.html#Handler(android.os.Looper))

You are also not required to return to the main thread explicitly:
![scr](https://cloud.githubusercontent.com/assets/3272235/26666791/86e58fca-46aa-11e7-9eb4-6ab012f26485.png)
